pytorch_lightning/callbacks/pruning.py

```diff
@@ -282,33 +288,35 @@ def apply_lottery_ticket_hypothesis(self):
         The ``resample_parameters`` argument can be used to reset the parameters with a new :math:`\theta_z \sim \mathcal{D}_\theta`
         """  # noqa: E501
 
-        def copy_param(new, old, name: str) -> None:
+        def copy_param(new: nn.Module, old: nn.Module, name: str) -> None:
             dst = getattr(new, name)
             src = getattr(old, name)
             if dst is None or src is None or not isinstance(dst, torch.Tensor) or not isinstance(src, torch.Tensor):
                 return
             dst.data = src.data.to(dst.device)
 
+        assert self._original_layers is not None
         for d in self._original_layers.values():
-            copy, names = d["data"], d["names"]
-            if self._resample_parameters and hasattr(copy, "reset_parameters"):
+            copy = d['data']
+            names = d["names"]
```

Comment:
```suggestion
            copy = d["data"]
            names = d["names"]
```

Reply:
shameless plug again for using the black formatter so we don't need to comment about quotation mark styles in PR reviews @Borda 😄

Reply:
+1